docs/docs/sample-scripts/entra/entra-remove-oauth2-permission-grants/index.mdx

```diff
@@ -0,0 +1,121 @@
+    It can filter and remove permissions based on how they were granted - either by admin consent, user consent, or both.
+
+  .PARAMETER appRegistrationName
+    The display name of the application registration in Entra ID from which to remove permission grants.
+    This parameter is mandatory.
+
+  .PARAMETER GrantType
+    Specifies which type of permission grants to remove. Valid values are:
+    - 'By Admin': Removes only permissions granted through admin consent (consentType = AllPrincipals)
+    - 'By User': Removes only permissions granted through user consent (consentType != AllPrincipals)
+    - 'All': Removes both admin and user consented permissions
+    Default value is: 'Granted By User'
+
+  .EXAMPLE
+    Remove-UserPermissionsGrants -appRegistrationName "My App"
+    Removes all user-consented permission grants for the application "My App"
+
+  .EXAMPLE
+    Remove-UserPermissionsGrants -appRegistrationName "My App" -GrantType "All"
+    Removes all permission grants (both admin and user consented) for the application "My App"
+
+  .EXAMPLE
+    Remove-UserPermissionsGrants -appRegistrationName "My App" -GrantType "By Admin"
+    Removes only admin-consented permission grants for the application "My App"
+
+  .NOTES
+    Prerequisites:
+    - Microsoft 365 CLI (m365) must be installed and configured
+    - User must have appropriate permissions in Entra ID to manage OAuth2 permission grants (e.g. Global Admin)
+
+  .LINK
+    https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/manage-application-permissions
+  #>
+
+  function Remove-PermissionsGrants {
+    param(
+      [Parameter(Mandatory = $true)]
+      [string]$appRegistrationName,
+
+      [Parameter(Mandatory = $false)]
+      [ValidateSet('By Admin', 'By User', 'All')] $GrantType = 'By User'
+    )
+    
+    Clear-Host
+    Write-Host "Removing permission grants for the app registration '$appRegistrationName'"
+    Write-Host "Selected grant type: '$GrantType'`n"
+
+    # Get Service Principal that represents the app registration
+    $app = m365 entra enterpriseapp list --query "[?appDisplayName == '$($appRegistrationName)']" | ConvertFrom-Json
+
+    # Filter (delegated) permissions granted with according consent for the given service principal
+    $selectedPermissionsGrants = @()
+    if ($GrantType -eq 'All') {
+      $selectedPermissionsGrants = m365 entra oauth2grant list --spObjectId $app.id | ConvertFrom-Json
+    }
+    elseif ($GrantType -eq 'Granted By Admin') {
+      $selectedPermissionsGrants = m365 entra oauth2grant list --spObjectId $app.id --query "[?consentType == 'AllPrincipals']" | ConvertFrom-Json
+    }
+    else {
+      # granted by user
+      $selectedPermissionsGrants = m365 entra oauth2grant list --spObjectId $app.id --query "[?consentType != 'AllPrincipals']" | ConvertFrom-Json
+    }
+
+    Write-Host "👉 $($selectedPermissionsGrants.Length) definitions found"
```

Comment:
Let's write an empty line below this one, it makes for better readability...

screenshot from my end: